api/src/main/java/org/openmrs/api/db/hibernate/HibernateConditionDAO.java

```diff
@@ -0,0 +1,104 @@
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.api.db.hibernate;
+
+import java.util.List;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.hibernate.Query;
+import org.hibernate.SessionFactory;
+import org.openmrs.Condition;
+import org.openmrs.Patient;
+import org.openmrs.api.db.ConditionDAO;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * Hibernate implementation of the ConditionDAO
+ *
+ * @see ConditionDAO
+ * @see org.openmrs.api.ConditionService
+ */
+public class HibernateConditionDAO implements ConditionDAO {
+	
+	private static final Log log = LogFactory.getLog(ConditionDAO.class);
+	
+	/**
+	 * Hibernate session factory
+	 */
+	private SessionFactory sessionFactory;
+	
+	/**
+	 * Set session factory
+	 *
+	 * @param sessionFactory
+	 */
+	public void setSessionFactory(SessionFactory sessionFactory) {
+		this.sessionFactory = sessionFactory;
+	}
+	
+	/**
+	 * Saves the condition.
+	 *
+	 * @param condition the condition to save.
+	 * @return the saved condition.
+	 */
+	@Override
+	@Transactional
+	public Condition saveCondition(Condition condition) {
+		sessionFactory.getCurrentSession().saveOrUpdate(condition);
+		return condition;
+	}
+	
+	/**
+	 * Gets the condition attached to the specified UUID.
+	 *
+	 * @param uuid the uuid to search for in the database.
+	 * @return the condition associated with the UUID.
+	 */
+	@Override
+	@Transactional(readOnly = true)
+	public Condition getConditionByUuid(String uuid) {
+		return (Condition) sessionFactory.getCurrentSession().createQuery("from Condition c where c.uuid = :uuid")
+				.setString("uuid", uuid).uniqueResult();
+	}
+	
+	/**
+	 * Gets all conditions related to the specified patient.
+	 *
+	 * @param patient the patient whose condition history is being queried.
+	 * @return all active and non active conditions related to the specified patient.
+	 */
+	@Override
+	@Transactional(readOnly = true)
+	public List<Condition> getConditionHistory(Patient patient) {
+		Query query = sessionFactory.getCurrentSession().createQuery(
+				"select con from Condition as con where con.patient.patientId = :patientId and con.voided = false " +
+						"order by con.dateCreated desc");
+		query.setInteger("patientId", patient.getId());
+		return query.list();
+	}
+	
+	/**
+	 * Gets all active conditions related to the specified patient.
+	 *
+	 * @param patient the patient whose active conditions are being queried.
+	 * @return all active conditions associated with the specified patient.
+	 */
+	@Override
+	@Transactional(readOnly = true)
+	public List<Condition> getActiveConditions(Patient patient) {
```

Comment:
Can you add a method for deleting a condition?

Reply:
okay